packages/network-controller/tests/provider-api-tests/shared-tests.ts

```diff
@@ -106,48 +116,51 @@ export const testsForProviderType = (providerType: ProviderType) => {
 
     describe('methods that assume there is no block param', () => {
       const assumingNoBlockParam = [
-        // { name: 'eth_getFilterLogs', numberOfParameters: 1 },
-        // { name: 'eth_blockNumber', numberOfParameters: 0 },
+        { name: 'eth_blockNumber', numberOfParameters: 0 },
         { name: 'eth_estimateGas', numberOfParameters: 2 },
         { name: 'eth_gasPrice', numberOfParameters: 0 },
         { name: 'eth_getBlockByHash', numberOfParameters: 2 },
+        // NOTE: eth_getBlockTransactionCountByNumber does take a block param at
+        // the 0th index, but this is not handled by our cache middleware
+        // currently
+        {
+          name: 'eth_getBlockTransactionCountByNumber',
+          numberOfParameters: 1,
+        },
+        // NOTE: eth_getTransactionByBlockNumberAndIndex does take a block param
+        // at the 0th index, but this is not handled by our cache middleware
+        // currently
+        {
+          name: 'eth_getTransactionByBlockNumberAndIndex',
+          numberOfParameters: 2,
+        },
         {
           name: 'eth_getBlockTransactionCountByHash',
           numberOfParameters: 1,
         },
+        { name: 'eth_getFilterLogs', numberOfParameters: 1 },
         {
           name: 'eth_getTransactionByBlockHashAndIndex',
           numberOfParameters: 2,
         },
         { name: 'eth_getUncleByBlockHashAndIndex', numberOfParameters: 2 },
+        // NOTE: eth_getUncleByBlockNumberAndIndex does take a block param at
+        // the 0th index, but this is not handled by our cache middleware
+        // currently
+        { name: 'eth_getUncleByBlockNumberAndIndex', numberOfParameters: 2 },
         { name: 'eth_getUncleCountByBlockHash', numberOfParameters: 1 },
-      ];
-
-      // NOTE: these methods do take a block param
-      // but this is not handled by our cache middleware currently
-      const blockParamIgnored = [
+        // NOTE: eth_getUncleCountByBlockNumber does take a block param at the
+        // 0th index, but this is not handled by our cache middleware currently
         { name: 'eth_getUncleCountByBlockNumber', numberOfParameters: 1 },
-        { name: 'eth_getUncleByBlockNumberAndIndex', numberOfParameters: 2 },
-        {
-          name: 'eth_getTransactionByBlockNumberAndIndex',
-          numberOfParameters: 2,
-        },
-        {
-          name: 'eth_getBlockTransactionCountByNumber',
-          numberOfParameters: 1,
-        },
       ];
-
-      assumingNoBlockParam
-        .concat(blockParamIgnored)
```

Comment:
I've copied `shared-tests.ts` from the extension into this repo since the behavior we want to test now should be exactly the same. This should address this among others.